instawork_app/views.py

```diff
@@ -32,75 +33,74 @@ def index(request):
 
     return render(request, 'instawork_app/index.html', context)
 
-def add(request): 
+def add(request):
 
-    context = {
-        'is_add_form': True
-    }
+    if request.method == 'POST':
+        form = MemberForm(request.POST)
 
-    if request.method == 'POST': 
-        name = request.POST.get('name')
-        position = request.POST.get('position')
-        phone = request.POST.get('phone')
-        email = request.POST.get('email')
+        if form.is_valid():
+            new_member_id = len(request.session.get('members', [])) + 1
+            new_member = {
+                'id': new_member_id,
+                'name': form.cleaned_data['name'],
+                'surname': form.cleaned_data['surname'],
+                'phone': form.cleaned_data['phone'],
+                'email': form.cleaned_data['email'],
+                'role': form.cleaned_data['role'],
+            }
 
+            members = request.session.get('members', [])
+            members.append(new_member)
+            request.session['members'] = members
 
-        members = request.session.get('members', [])
-        new_member_id = len(members) + 1
-        new_member = {
-            'id': new_member_id, # TODO: fix this bug 
-            'name': name,
-            'position': position,
-            'phone': phone,
-            'email': email,
-        }
-        print('member [new member]', new_member)
-        print('members [add]:', len(members))
+            return redirect('index')
+    else:
+        form = MemberForm()  
+
+    context = {
+        'is_add_form': True,
+        'form': form
+    }
 
-        members.append(new_member)
-        request.session['members'] = members
-        print('members [new member]', members)
-        
-        return redirect('index')
-    
-    # TODO: extract the add_member content into a component and then call it in each of the forms
     return render(request, 'add_member/index.html', context)
 
-def edit(request, member_index=None): 
+def edit(request, member_index=None):
```

Comment:
![image](https://github.com/Dunridge/instawork-task/assets/29373357/aa3edda8-881d-43cc-9d5a-ec79c7ad0c6d)

Reply:
Got this error when editing a memeber that has been edited